src/Adapter/Currency/CommandHandler/ToggleExchangeRateAutomatizationHandler.php

```diff
@@ -0,0 +1,262 @@
+use PrestaShop\PrestaShop\Core\Domain\Currency\Command\ToggleExchangeRateAutomatizationCommand;
+use PrestaShop\PrestaShop\Core\Domain\Currency\CommandHandler\ToggleExchangeRateAutomatizationHandlerInterface;
+use PrestaShop\PrestaShop\Core\Domain\Currency\Exception\ScheduleExchangeRatesUpdateException;
+use PrestaShop\PrestaShop\Core\Domain\Currency\Exception\CurrencyException;
+use PrestaShopException;
+use Shop;
+use Symfony\Component\Translation\TranslatorInterface;
+use Tools;
+
+/**
+ * Class ToggleExchangeRateAutomatizationHandler is responsible for turning on or off the setting - if its on then
+ * in CronJobs module it creates new record with url which points to the script which is being executed at certain time
+ * of period. If the setting is off then it removes that record.
+ *
+ * @todo: an issue with multi-store cron task scheduler.
+ *
+ * @internal
+ */
+final class ToggleExchangeRateAutomatizationHandler implements ToggleExchangeRateAutomatizationHandlerInterface
+{
+    /**
+     * @var Configuration
+     */
+    private $configuration;
+
+    /**
+     * @var Shop
+     */
+    private $contextShop;
+
+    /**
+     * @var bool
+     */
+    private $isCronJobModuleInstalled;
+
+    /**
+     * @var ShopUrlDataProvider
+     */
+    private $shopUrlDataProvider;
+
+    /**
+     * @var string
+     */
+    private $adminBaseUrl;
+
+    /**
+     * @var string
+     */
+    private $dbPrefix;
+
+    /**
+     * @var TranslatorInterface
+     */
+    private $translator;
+
+    /**
+     * @param Configuration $configuration
+     * @param Shop $contextShop
+     * @param ShopUrlDataProvider $shopUrlDataProvider
+     * @param TranslatorInterface $translator
+     * @param bool $isCronJobModuleInstalled
+     * @param string $adminBaseUrl
+     * @param string $dbPrefix
+     */
+    public function __construct(
+        Configuration $configuration,
+        Shop $contextShop,
+        ShopUrlDataProvider $shopUrlDataProvider,
+        TranslatorInterface $translator,
+        $isCronJobModuleInstalled,
+        $adminBaseUrl,
+        $dbPrefix
+    ) {
+        $this->configuration = $configuration;
+        $this->contextShop = $contextShop;
+        $this->isCronJobModuleInstalled = $isCronJobModuleInstalled;
+        $this->shopUrlDataProvider = $shopUrlDataProvider;
+        $this->adminBaseUrl = $adminBaseUrl;
+        $this->dbPrefix = $dbPrefix;
+        $this->translator = $translator;
+    }
+
+    /**
+     * {@inheritdoc}
+     *
+     * @throws CurrencyException
+     */
+    public function handle(ToggleExchangeRateAutomatizationCommand $command)
+    {
+        if (!$this->isCronJobModuleInstalled) {
+            throw new ScheduleExchangeRatesUpdateException(
+                'Live exchange rates feature cannot be modified because "cronjob" module is not installed',
+                ScheduleExchangeRatesUpdateException::CRON_TASK_MANAGER_MODULE_NOT_INSTALLED
```

Comment:
i think you should either use `Schedule` or `Automate` (following your handler name).